src/lib/falcor/falcorUtils.js

```diff
@@ -28,3 +28,73 @@ export function setAppReady() {
   console.log("APP SET AS READY")
   appReady = true
 }
+
+// Below function based on function from: https://github.com/ekosz/falcor-expand-cache
+// MIT License
+// Copyright (c) 2015 Eric Koslow
+export function expandCache(cache) {
+  function followPath(path) {
+    // If using dot notation obj.key.key.key
+    if (typeof path === "string") {
+      path = path.split('.');
+    }
+    return path.reduce((acc, part) => {
+      if (acc !== undefined && acc.hasOwnProperty(part)) {
+        return acc[part];
+      }
+
+      return undefined;
+    }, cache);
+  }
+
+  function isObject(val) {
+    if (val === null || val instanceof Array) return false;
+    return typeof val === "object";
+  }
+
+  // If empty return itself
+  if (!cache) return cache;
+  // Expanding
+  let stack = [];
+  Object.keys(cache).forEach((key) => {
+    stack.push([key]);
+  })
+  while (stack.length > 0) {
+    let pathArray = stack.pop();
+    // Parent also works for array length 1, aka initial keys
+    // Used for assigning
+    let parent = followPath(pathArray.slice(0, pathArray.length-1));
```

Comment:
True, but I always only put arrays into the stack, so I know what I get out when I pop from it.

Also, the check I put into followPath was from when I was considering using string paths, and I just left it there if someone else ever wanted to use string paths, I'll put in a comment saying that it's currently redundant, but I think it's okay keeping it there right? No need to delete the check, it just gives flexibility if anyone wanted to use it later.